src/map/lua/lua_baseentity.cpp

```diff
@@ -6620,6 +6621,117 @@ inline int32 CLuaBaseEntity::getMissionLogEx(lua_State *L)
     return 1;
 }
 
+/************************************************************************
+*  Function: setEminenceCompleted()
+*  Purpose :
+*  Example : player:setEminenceCompleted(1)
+*  Notes   : optional arg 2 can set completion state explicitly (1/0)
+************************************************************************/
+
+inline int32 CLuaBaseEntity::setEminenceCompleted(lua_State *L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+
+    CCharEntity* PChar = (CCharEntity*)m_PBaseEntity;
+    uint16 recordID = (uint16)lua_tointeger(L, 1);
+
+    bool status;
+    if (lua_gettop(L) < 2)
+        status = true;
+    else
+        status = lua_tointeger(L, 2) != 0;
+
+    charutils::DelEminenceRecord(PChar, recordID);
+    charutils::SetEminenceRecordCompletion(PChar, recordID, status);
+
+    return 0;
+}
+
+/************************************************************************
+*  Function: getEminenceCompleted()
+*  Purpose : Returns true if eminence is flagged complete for player
+*  Example : player:getEminenceCompleted(1)
```

Comment:
I mentioned this in Discord, and I think I saw someone else mention it, is this 1 going to get tabled later?
`player:getEminenceCompleted(1)`

Reply:
There wasn't a real consensus on it right now; there's going to be 1300 trials, and they'll mostly be referenced algorithmically once the check system goes in so I'm not sure how much value there is there. Hunts and Regimes don't use enums, and they work similarly too. At any rate, something to keep in mind for phase 2!